src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Helpers/PolicyHelper.cs

```diff
@@ -41,5 +45,133 @@ public static List<string> GetRolesWithAccess(XacmlPolicy policy)
 
             return roleCodes.ToList();
         }
+
+        /// <summary>
+        /// Gets the entire policy as a list of <see cref="ResourcePolicy"/>. 
+        /// </summary>
+        /// <param name="policy">The policy</param>
+        /// <returns>List of resource policies</returns>
+        public static List<ResourcePolicy> GetResourcePoliciesFromXacmlPolicy(XacmlPolicy policy)
+        {
+            Dictionary<string, ResourcePolicy> resourcePolicies = new Dictionary<string, ResourcePolicy>();
+
+            foreach (XacmlRule rule in policy.Rules)
+            {
+                if (rule.Effect.Equals(XacmlEffectType.Permit) && rule.Target != null)
+                {
+                    List<string> policyKeys = new List<string>();
+                    List<RoleGrant> roles = new List<RoleGrant>();
+                    List<ResourceAction> actions = new List<ResourceAction>();
+                    foreach (XacmlAnyOf anyOf in rule.Target.AnyOf)
+                    {
+                        foreach (XacmlAllOf allOf in anyOf.AllOf)
```

Comment:
I am unsure of this loop. It seems for me it would include part of AllOfResources also when there are parts that is not matching.